pkg/scraper/client/resource/decode.go

```diff
@@ -51,6 +52,8 @@ func decodeBatch(b []byte, defaultTime time.Time, nodeName string) *storage.Metr
 		if et, err = parser.Next(); err != nil {
```

Comment:
Could you provide an example of error that would be returned here?

I would like to make sure that reporting just error is enough for users to understand what is a problem. I would consider looking into parser state to extract line that we causes the problem and add it to error.

Reply:
A response error will look like the below.
```
failed parsing metrics: "INVALID" is not a valid start token
```

Reply:
That is the most ambiguous error message on `promparse`. Should I append the `series` into the error message?
https://github.com/prometheus/prometheus/blob/v2.5.0/pkg/textparse/promparse.go#L360

```diff
- return nil, fmt.Errorf("failed parsing metrics: %w", err)
+ series, _, _ := parser.Series()
+ return nil, fmt.Errorf("failed parsing metrics `%s`: %w", series, err)
```

More useful information for debugging 😅
```
failed parsing metrics `container_start_time_seconds{container="metrics-server",namespace="kubernetes-dashboard",pod="kubernetes-dashboard-metrics-server-77db45cdf4-fppzx"}`: "INVALID" is not a valid start token
```

Reply:
Providing name of the series doesn't seem useful here. If we encountered ephemeral incorrect values, then knowing what container/node is responsible is not enough to understand what is the issue. I don't think it makes sense to include debug information if we cannot report the value that was invalid.